aws/resource_aws_ec2_capacity_reservation_test.go

```diff
@@ -0,0 +1,222 @@
+	})
+}
+
+func testSweepEc2CapacityReservations(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).ec2conn
+
+	resp, err := conn.DescribeCapacityReservations(&ec2.DescribeCapacityReservationsInput{})
+
+	if err != nil {
+		return fmt.Errorf("Error retrieving capacity reservations: %s", err)
+	}
+
+	if len(resp.CapacityReservations) == 0 {
+		log.Print("[DEBUG] No capacity reservations to sweep")
+		return nil
+	}
+
+	for _, r := range resp.CapacityReservations {
+		if *r.State != "cancelled" {
+			id := aws.StringValue(r.CapacityReservationId)
+
+			log.Printf("[INFO] Cancelling capacity reservation EC2 Instance: %s", id)
+
+			opts := &ec2.CancelCapacityReservationInput{
+				CapacityReservationId: aws.String(id),
+			}
+
+			_, err := conn.CancelCapacityReservation(opts)
+
+			if err != nil {
+				log.Printf("[ERROR] Error cancelling capacity reservation (%s): %s", id, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSEc2CapacityReservation_importBasic(t *testing.T) {
+	resourceName := "aws_ec2_capacity_reservation.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEc2CapacityReservationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEc2CapacityReservationConfig,
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSEc2CapacityReservation_basic(t *testing.T) {
+	var cr ec2.CapacityReservation
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEc2CapacityReservationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEc2CapacityReservationConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEc2CapacityReservationExists("aws_ec2_capacity_reservation.default", &cr),
+					resource.TestCheckResourceAttr("aws_ec2_capacity_reservation.default", "instance_type", "t2.micro"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSEc2CapacityReservation_endDate(t *testing.T) {
+	var cr ec2.CapacityReservation
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEc2CapacityReservationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEc2CapacityReservationConfig_endDate,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEc2CapacityReservationExists("aws_ec2_capacity_reservation.default", &cr),
+					resource.TestCheckResourceAttr("aws_ec2_capacity_reservation.default", "end_date", "2019-10-31T07:39:57Z"),
+					resource.TestCheckResourceAttr("aws_ec2_capacity_reservation.default", "end_date_type", "limited"),
+				),
+			},
```

Comment:
This test should include `TestStep` to verify updating the end date and/or end date type.